### PR TITLE
Use whitelist for npm package instead of .npmignore

### LINKED ISSUE #322: Problem with publishing npm package
## Requirement - what kind of business use case are you trying to solve?
Publish npm package 

## Problem - what in Jaeger blocks you from solving the requirement?
Missed folder `dist/src` in npm package after the update of `.npmingore` #307

## Proposal - what do you suggest to solve the problem or improve the existing situation?
Fix `.nmpignore` or use [whitelist](https://docs.npmjs.com/files/package.json#files) instead of blacklist

### PR DESCRIPTION
**Pull request by yepninja**
Resolves #322

### CODE REVIEW

**Comment by yepninja** [comment]
```
npm pack --dry-run
npm notice
npm notice 📦  jaeger-client@3.13.0
npm notice === Tarball Contents ===
npm notice 3.2kB  package.json
npm notice 4.9kB  CHANGELOG.md
npm notice 11.4kB LICENSE
npm notice 15.3kB README.md
npm notice 283B   dist/src/_flow/baggage_restriction_manager.js
npm notice 1.3kB  dist/src/_flow/baggage_restriction_manager.js.map
npm notice 47B    dist/src/_flow/bufrw.js
npm notice 89B    dist/src/_flow/bufrw.js.map
npm notice 224B   dist/src/_flow/context.js
npm notice 864B   dist/src/_flow/context.js.map
npm notice 51B    dist/src/_flow/crossdock.js
npm notice 93B    dist/src/_flow/crossdock.js.map
npm notice 51B    dist/src/_flow/extractor.js
npm notice 93B    dist/src/_flow/extractor.js.map
npm notice 260B   dist/src/_flow/injector.js
npm notice 892B   dist/src/_flow/injector.js.map
npm notice 55B    dist/src/_flow/jaeger-thrift.js
npm notice 97B    dist/src/_flow/jaeger-thrift.js.map
npm notice 48B    dist/src/_flow/logger.js
npm notice 90B    dist/src/_flow/logger.js.map
npm notice 49B    dist/src/_flow/metrics.js
npm notice 91B    dist/src/_flow/metrics.js.map
npm notice 56B    dist/src/_flow/process_setter.js
npm notice 98B    dist/src/_flow/process_setter.js.map
npm notice 228B   dist/src/_flow/reporter.js
npm notice 1.2kB  dist/src/_flow/reporter.js.map
npm notice 56B    dist/src/_flow/sampler-thrift.js
npm notice 98B    dist/src/_flow/sampler-thrift.js.map
npm notice 49B    dist/src/_flow/sampler.js
npm notice 91B    dist/src/_flow/sampler.js.map
npm notice 48B    dist/src/_flow/socket.js
npm notice 90B    dist/src/_flow/socket.js.map
npm notice 51B    dist/src/_flow/throttler.js
npm notice 93B    dist/src/_flow/throttler.js.map
npm notice 258B   dist/src/_flow/tracer.js
npm notice 1.0kB  dist/src/_flow/tracer.js.map
npm notice 4.1kB  dist/src/baggage/baggage_setter.js
npm notice 5.0kB  dist/src/baggage/baggage_setter.js.map
npm notice 2.4kB  dist/src/baggage/default_baggage_restriction_manager.js
npm notice 1.8kB  dist/src/baggage/default_baggage_restriction_manager.js.map
npm notice 2.1kB  dist/src/baggage/restriction.js
npm notice 1.6kB  dist/src/baggage/restriction.js.map
npm notice 7.1kB  dist/src/configuration_env.js
npm notice 11.0kB dist/src/configuration_env.js.map
npm notice 11.0kB dist/src/configuration.js
npm notice 14.0kB dist/src/configuration.js.map
npm notice 3.8kB  dist/src/constants.js
npm notice 4.5kB  dist/src/constants.js.map
npm notice 1.8kB  dist/src/default_context.js
npm notice 1.1kB  dist/src/default_context.js.map
npm notice 4.4kB  dist/src/index.js
npm notice 3.2kB  dist/src/index.js.map
npm notice 53B    dist/src/jaeger-idl/.gitignore
npm notice 99B    dist/src/jaeger-idl/.travis.yml
npm notice 1.1kB  dist/src/jaeger-idl/LICENSE
npm notice 848B   dist/src/jaeger-idl/Makefile
npm notice 222B   dist/src/jaeger-idl/README.md
npm notice 1.4kB  dist/src/jaeger-idl/thrift/agent.thrift
npm notice 1.4kB  dist/src/jaeger-idl/thrift/aggregation_validator.thrift
npm notice 1.7kB  dist/src/jaeger-idl/thrift/baggage.thrift
npm notice 1.3kB  dist/src/jaeger-idl/thrift/crossdock/tracetest.thrift
npm notice 1.7kB  dist/src/jaeger-idl/thrift/dependency.thrift
npm notice 3.5kB  dist/src/jaeger-idl/thrift/jaeger.thrift
npm notice 2.9kB  dist/src/jaeger-idl/thrift/sampling.thrift
npm notice 11.9kB dist/src/jaeger-idl/thrift/zipkincore.thrift
npm notice 1.7kB  dist/src/logger.js
npm notice 988B   dist/src/logger.js.map
npm notice 3.3kB  dist/src/metrics/metrics.js
npm notice 6.9kB  dist/src/metrics/metrics.js.map
npm notice 1.7kB  dist/src/metrics/noop/counter.js
npm notice 966B   dist/src/metrics/noop/counter.js.map
npm notice 1.7kB  dist/src/metrics/noop/gauge.js
npm notice 955B   dist/src/metrics/noop/gauge.js.map
npm notice 2.4kB  dist/src/metrics/noop/metric_factory.js
npm notice 1.5kB  dist/src/metrics/noop/metric_factory.js.map
npm notice 1.7kB  dist/src/metrics/noop/timer.js
npm notice 955B   dist/src/metrics/noop/timer.js.map
npm notice 4.5kB  dist/src/metrics/prometheus.js
npm notice 5.2kB  dist/src/metrics/prometheus.js.map
npm notice 1.5kB  dist/src/propagators/baggage.js
npm notice 2.0kB  dist/src/propagators/baggage.js.map
npm notice 1.8kB  dist/src/propagators/binary_codec.js
npm notice 1.1kB  dist/src/propagators/binary_codec.js.map
npm notice 5.6kB  dist/src/propagators/text_map_codec.js
npm notice 7.0kB  dist/src/propagators/text_map_codec.js.map
npm notice 8.6kB  dist/src/propagators/zipkin_b3_text_map_codec.js
npm notice 11.0kB dist/src/propagators/zipkin_b3_text_map_codec.js.map
npm notice 2.8kB  dist/src/rate_limiter.js
npm notice 3.1kB  dist/src/rate_limiter.js.map
npm notice 2.9kB  dist/src/reporters/composite_reporter.js
npm notice 2.8kB  dist/src/reporters/composite_reporter.js.map
npm notice 5.9kB  dist/src/reporters/http_sender.js
npm notice 7.9kB  dist/src/reporters/http_sender.js.map
npm notice 2.6kB  dist/src/reporters/in_memory_reporter.js
npm notice 2.1kB  dist/src/reporters/in_memory_reporter.js.map
npm notice 2.4kB  dist/src/reporters/logging_reporter.js
npm notice 1.8kB  dist/src/reporters/logging_reporter.js.map
npm notice 2.2kB  dist/src/reporters/noop_reporter.js
npm notice 1.4kB  dist/src/reporters/noop_reporter.js.map
npm notice 4.7kB  dist/src/reporters/remote_reporter.js
npm notice 5.7kB  dist/src/reporters/remote_reporter.js.map
npm notice 2.2kB  dist/src/reporters/sender_utils.js
npm notice 2.0kB  dist/src/reporters/sender_utils.js.map
npm notice 7.8kB  dist/src/reporters/udp_sender.js
npm notice 11.3kB dist/src/reporters/udp_sender.js.map
npm notice 3.0kB  dist/src/samplers/const_sampler.js
npm notice 2.5kB  dist/src/samplers/const_sampler.js.map
npm notice 5.6kB  dist/src/samplers/guaranteed_throughput_sampler.js
npm notice 6.0kB  dist/src/samplers/guaranteed_throughput_sampler.js.map
npm notice 5.7kB  dist/src/samplers/per_operation_sampler.js
npm notice 6.5kB  dist/src/samplers/per_operation_sampler.js.map
npm notice 3.4kB  dist/src/samplers/probabilistic_sampler.js
npm notice 3.1kB  dist/src/samplers/probabilistic_sampler.js.map
npm notice 4.2kB  dist/src/samplers/ratelimiting_sampler.js
npm notice 4.3kB  dist/src/samplers/ratelimiting_sampler.js.map
npm notice 9.2kB  dist/src/samplers/remote_sampler.js
npm notice 12.6kB dist/src/samplers/remote_sampler.js.map
npm notice 10.1kB dist/src/span_context.js
npm notice 13.1kB dist/src/span_context.js.map
npm notice 11.9kB dist/src/span.js
npm notice 15.1kB dist/src/span.js.map
npm notice 10.2kB dist/src/tchannel_bridge.js
npm notice 13.2kB dist/src/tchannel_bridge.js.map
npm notice 3.6kB  dist/src/test_util.js
npm notice 4.0kB  dist/src/test_util.js.map
npm notice 6.2kB  dist/src/thrift.js
npm notice 8.6kB  dist/src/thrift.js.map
npm notice 1.2kB  dist/src/thriftrw-idl/agent.thrift
npm notice 2.1kB  dist/src/throttler/default_throttler.js
npm notice 1.6kB  dist/src/throttler/default_throttler.js.map
npm notice 7.6kB  dist/src/throttler/remote_throttler.js
npm notice 10.4kB dist/src/throttler/remote_throttler.js.map
npm notice 15.9kB dist/src/tracer.js
npm notice 21.2kB dist/src/tracer.js.map
npm notice 6.6kB  dist/src/util.js
npm notice 8.5kB  dist/src/util.js.map
npm notice === Tarball Details ===
npm notice name:          jaeger-client
npm notice version:       3.13.0
npm notice filename:      jaeger-client-3.13.0.tgz
npm notice package size:  106.4 kB
npm notice unpacked size: 532.2 kB
npm notice shasum:        f2f8a442074ebbaef951123c5d4227c3089ddae6
npm notice integrity:     sha512-6dHBXZ95vHgPh[...]1nwu34PXjI4Vw==
npm notice total files:   136
npm notice
jaeger-client-3.13.0.tgz
```

File: package.json [hunk 1]
```diff
@@ -13,7 +13,13 @@
     "type": "git",
     "url": "git://github.com/jaegertracing/jaeger-client-node.git"
   },
-  "contributors": ["Yuri Shkuro <ys@uber.com>", "Won Jun Jang <wjang@uber.com>"],
+  "files": [
+    "dist/src"
+  ],
```
Comment (yurishkuro) [inline package.json]: is this the whitelist?

Comment (yepninja) [reply package.json]: According to [npm docs](https://docs.npmjs.com/files/package.json#files), LICENSE and README (also CHANGELOG, NOTICE, package.json, and main field) are always included.

Comment (yurishkuro) [reply package.json]: +1

Comment (yurishkuro) [reply package.json]: > the correct pattern was without dist: `src/**`

or may be not - Won is still investigating

**Pull request merged**